node/pkg/governor/governor_prices.go

```diff
@@ -26,42 +26,82 @@ import (
 // The CoinGecko API is documented here: https://www.coingecko.com/en/api/documentation
 // An example of the query to be generated: https://api.coingecko.com/api/v3/simple/price?ids=gemma-extending-tech,bitcoin,weth&vs_currencies=usd
 
+// coinGeckoQueryIntervalInMins specifies how often we query CoinGecko for prices.
 const coinGeckoQueryIntervalInMins = 15
 
+// tokensPerCoinGeckoQuery specifies how many tokens will be in each CoinGecko query. The token list will be broken up into chunks of this size.
+const tokensPerCoinGeckoQuery = 200
+
+// initCoinGecko builds the set of CoinGecko queries that will be used to update prices. It also starts a go routine to periodically do the queries.
 func (gov *ChainGovernor) initCoinGecko(ctx context.Context, run bool) error {
+	// Create a slice of all the CoinGecko IDs so we can create the corresponding queries.
+	ids := make([]string, 0, len(gov.tokensByCoinGeckoId))
+	for id := range gov.tokensByCoinGeckoId {
+		ids = append(ids, id)
+	}
+
+	// Create the set of queries, breaking the IDs into the appropriate size chunks.
+	gov.coinGeckoQueries = createCoinGeckoQueries(ids, tokensPerCoinGeckoQuery)
+	for queryIdx, query := range gov.coinGeckoQueries {
+		gov.logger.Info("cgov: coingecko query: ", zap.Int("queryIdx", queryIdx), zap.String("query", query))
+	}
+
+	if len(gov.coinGeckoQueries) == 0 {
+		gov.logger.Info("cgov: did not find any tokens, nothing to do!")
+		return nil
```

Comment:
Should this hard error when we cannot get prices or is this ok with this because there is a floor price to fall back on?

Reply:
This is the way it has been since the beginning. I think we were okay with it because of the floor prices. If CG ever becomes unavailable, we may want to be able to continue. But I can make it exit if you prefer. Let me know.